src/query/functions/linear/datetime.go

```diff
@@ -71,37 +81,30 @@ var (
 	}
 )
 
-// NewDateOp creates a new date op based on the type
-func NewDateOp(optype string) (BaseOp, error) {
-	if _, ok := datetimeFuncs[optype]; !ok {
-		return emptyOp, fmt.Errorf("unknown date type: %s", optype)
+func buildTransform(args []interface{}, fn timeFn) block.ValueTransform {
```

Comment:
Hm is args used in this method at all?  Seems like only checked if length == 0?

Reply:
Yeah that's pretty much it; the way this family of functions work is that you can call them with 0 or 1 args, but either way it doesn't actually care about the value of the arg, just that it's there. I'll refactor this a bit to make it a bit nicer and clearer though (this will also address your comment re: `n.Func.Name != linear.RoundType`)